src/utils/EventUtils.cpp

```diff
@@ -230,10 +234,11 @@ void EventRegistry::finalize()
   for (auto & e : storedEvents)
     e.second.stop();
 
-  if (initialized) // this makes only sense when it was properly initialized
+  if (initialized && not precice::testMode) // this makes only sense when it was properly initialized
```

Comment:
It seems a bit odd that this is necessary.

I would rather refactor the events library and the testing framework instead of integrating hacks like these.

Reply:
Yes, it is odd. @fsimonis and me just agreed to fix this not here, but in #597 
I'll add a TODO in the sources.